### ISSUE TITLE
"error: unexpected end of file" in some ChildProcess

### ISSUE THREAD

**Issue by siko1056** [coala/coala-atom#21]
Dear maintainers,

I continued working on an atom project, containing a .coafile, which one month ago resulted in nice push notifications about too long lines etc. Now there comes always this error push notification:

```
Error: /bin/sh: mc: line 1: syntax error: unexpected end of file /bin/sh: error importing function definition for `mc'
Error: /bin/sh: mc: line 1: syntax error: unexpected end of file
/bin/sh: error importing function definition for `mc'
    at ChildProcess.<anonymous> (<home>/.atom/packages/coala/node_modules/atom-linter/node_modules/sb-exec/lib/index.js:36:20)
    at emitTwo (events.js:87:13)
    at ChildProcess.emit (events.js:172:7)
    at maybeClose (internal/child_process.js:818:16)
    at Socket.<anonymous> (internal/child_process.js:319:11)
    at emitOne (events.js:77:13)
    at Socket.emit (events.js:169:7)
    at Pipe._onclose (net.js:469:12)
```

My .coafile for Matlab files:

```
[Default]
bears = SpaceConsistencyBear, LineLengthBear

max_line_length = 80
use_spaces = True

files = *.m
```

Calling coala from command line, from within the project root, runs without any problems, gives the expected hints for improvement. A guess from my side, linter has had an update since the last release of coala-atom, maybe something changed?

For your information:
- I have installed coala 0.5.2
- atom@1.7.2 runs coala@0.2.0 and linter@1.11.4

Thank you for any kind of help, I can respond tomorrow first,
Kai

**Comment by sils** [comment]
Hi Kai, I'm CCing @AbdealiJK , our main plugin developer. We'll look into this, doing a release of coala 0.6 right now. (We're at http://linuxwochen.at/ in vienna right now so I can't look into it personally.)

**Comment by AbdealiLoKo** [comment]
More info:
With atom@1.5.3 , coala (atom plugin)@0.2.0, linter@1.11.3, coala@master, coala-bears@master
Bears tested: PEP8Bear, PyUnusedCodeBear, LineLengthBear, SpaceConsistencyBear
Tests: coala-atom tests pass

So, it does seem to be something with the latest atom.
I'm unable to download the latest atom for some reason (the connection keeps breaking).
Will update on this when I can.

It shouldn't be a problem with atom-linter (4.4.0 ->4.7.0) or linter (1.11.3 -> 1.11.4) as they only updated their minor versions. Possibly a problem with the atom (1.5.3->1.7.2)

@siko1056 Just to make sure, have you tried uninstalling and reinstalling the coala plugin ?

**Comment by AbdealiLoKo** [comment]
On further analysis it seems there was a bug in one of the versions of sb-exec between 1.0.0 and 1.0.4 which was fixed in 1.0.4 (https://github.com/steelbrain/atom-linter/commit/7271f9393d1d04cb80fe7d2aceaffa23114e4145)

Hence I'd recommend updating the dependencies of the coala atom plugin. This can be done by:
- reinstalling coala (which will force atom to reinstall latest versions of all deps)
- Use the "Update package dependencies: Update" item in the fuzzy searcher. (I've never used it myself though.)

**Comment by siko1056** [comment]
@AbdealiJK Yes, I have installed and uninstalled the coala and linter package a few times. Also "Update package dependencies: Update" item in the fuzzy searcher had no success.

```
.atom/packages/coala $ npm list
coala@0.2.0 /home/siko1056/.atom/packages/coala
├─┬ atom-linter@4.7.0
│ ├── named-js-regexp@1.3.1
│ ├─┬ sb-exec@1.0.5
│ │ ├── consistent-env@1.1.2
│ │ ├── extify@1.0.0
│ │ └─┬ sb-npm-path@1.0.0
│ │   └── sb-memoize@1.0.2
│ ├── sb-promisify@1.3.0
│ └─┬ tmp@0.0.28
│   └── os-tmpdir@1.0.1
└─┬ atom-package-deps@3.0.9
  └─┬ atom-package-path@1.1.0
    └── sb-callsite@1.1.2
```

A further investigation at this bug, after I have learned how to use the developer tools in atom and a bit of coffeescript, I decided to add to the points of rejection in

```
coala/node_modules/atom-linter/node_modules/sb-exec/lib/index.js
```

a `console.log` statement. Et voila, somehow the results of coala-format are there. But it gets lost the chain of function calls

```
coala -> atom-linter -> sb-exec -> NodeJS
```

which basically reduces to some simple NodeJS `require('child_process').spawn` command. And I am no longer sure, where the problem is located. I don't know NodeJS enough to say, if a specific version was the problem. I noticed, that installing coffescript on my system provided me another NodeJS, as atom seems to use (`process.version` in the developer console).

BTW, atom came out with 1.7.3, but the situation remains.
